src/Components/test/testassets/TestServer/ServerStartup.cs

```diff
@@ -23,6 +23,11 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc();
             services.AddServerSideBlazor();
             services.AddSingleton<ResourceRequestLog>();
+
+            // Since tests run in parallel, it's possible multiple servers will startup and read files being written by another test
+            // Use a unique directory per server to avoid this collision
+            services.AddDataProtection()
+                .PersistKeysToFileSystem(Directory.CreateDirectory(Path.GetRandomFileName()));
```

Comment:
@HaoK could we not have this use an ephemeral store?

Reply:
Yeah that probably would be easier, see https://github.com/dotnet/aspnetcore/blob/main/src/Identity/Specification.Tests/src/IdentitySpecificationTestBase.cs#L46

Reply:
Ah yeah, that makes more sense.

cc: @BrennanConroy in case you want to try this in the SignalR tests too.